### PR TITLE
add create and edit comment endpoints

### PR DESCRIPTION
**Pull request by AnnPostnikova**
https://docs.google.com/document/d/1upUYu5MaGjBW2MKP-wHM2hwef8t6TPGzpVNkF8njJDI/edit#heading=h.7jjwan5qecem

### CODE REVIEW

File: src/EventService.Broker/Requests/Interfaces/IUserService.cs [hunk 1]
```diff
@@ -9,7 +9,8 @@ namespace LT.DigitalOffice.EventService.Broker.Requests.Interfaces;
 [AutoInject]
 public interface IUserService
 {
-  Task<bool> CheckUsersExistenceAsync(List<Guid> usersIds, List<string> errors = null);
+  Task<bool> CheckUsersExistenceAsync(List<Guid> usersIds, List<string> errors = null);
+  Task<bool> CheckUserExistenceAsync(Guid userId, List<string> errors = null);
```
Comment (SolovyovDmitry) [inline src/EventService.Broker/Requests/Interfaces/IUserService.cs]: Is it required? Can't we just use list version?

File: src/EventService.Data/EventCommentRepository.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  public async Task<Guid?> CreateAsync(DbEventComment dbEventComment)
+  {
+    if (dbEventComment is null)
+    {
+      return null;
+    }
+
+    _provider.EventComments.Add(dbEventComment);
```
Comment (SolovyovDmitry) [inline src/EventService.Data/EventCommentRepository.cs:34]: why not async?

File: src/EventService.Data/EventCommentRepository.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    request.ApplyTo(dbEventComment);
+    dbEventComment.ModifiedBy = _httpContextAccessor.HttpContext.GetUserId();
+    dbEventComment.ModifiedAtUtc = DateTime.UtcNow;
+
+    await _provider.SaveAsync();
+
+    return true;
+  }
+
+  public async Task<bool> EditIsActiveAsync(Guid commentId, JsonPatchDocument<DbEventComment> request)
```
Comment (SolovyovDmitry) [inline src/EventService.Data/EventCommentRepository.cs:58]: can't is active be changed in previous method?

File: src/EventService.Mappers/Db/DbEventCommentMapper.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  {
+    return request is null
+      ? null
+      : new DbEventComment
+      {
+        Id = Guid.NewGuid(),
+        Content = request.Content,
+        UserId= request.UserId,
+        EventId= request.EventId,
+        ParentId= request.ParentId,
```
Comment (SolovyovDmitry) [inline src/EventService.Mappers/Db/DbEventCommentMapper.cs]: spaces

File: src/EventService.Models.Dto/Requests/EventComment/CreateEventCommentRequest.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace LT.DigitalOffice.EventService.Models.Dto.Requests.EventComment;
+
+public record CreateEventCommentRequest
+{
+  [Required]
+  [MaxLength(300)]
```
Comment (SolovyovDmitry) [inline src/EventService.Models.Dto/Requests/EventComment/CreateEventCommentRequest.cs]: leave it in validator

File: src/EventService.Validation/EventComment/CreateEventCommentRequestValidator.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+public class CreateEventCommentRequestValidator : AbstractValidator<CreateEventCommentRequest>, ICreateEventCommentRequestValidator
+{
+  public CreateEventCommentRequestValidator(
+    IEventRepository eventRepository,
+    IEventCommentRepository commentRepository,
+    IUserService userService)
+  {
+    RuleFor(x => x.EventId)
+      .Cascade(CascadeMode.Stop)
```
Comment (SolovyovDmitry) [inline src/EventService.Validation/EventComment/CreateEventCommentRequestValidator.cs]: this is obsolete. Use class and rule level properties as docs suggests

File: src/EventService.Data/EventCommentRepository.cs (hunk 3 above)
Comment (AnnPostnikova) [reply src/EventService.Data/EventCommentRepository.cs:58]: I think it's better to separate from each other

File: src/EventService.Data/EventCommentRepository.cs (hunk 2 above)
Comment (AnnPostnikova) [reply src/EventService.Data/EventCommentRepository.cs:34]: It is not necessary.
"This method is async only to allow special value generators, such as the one used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo', to access the database asynchronously. For all other cases the non async method should be used."

File: src/EventService.Business/Commands/EventComment/CreateEventCommentCommand.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    IDbEventCommentMapper mapper,
+    IResponseCreator responseCreator,
+    IHttpContextAccessor contextAccessor)
+  {
+    _repository = repository;
+    _mapper = mapper;
+    _validator = validator;
+    _responseCreator = responseCreator;
+    _contextAccessor = contextAccessor;
+  }
```
Comment (VLiamin) [inline src/EventService.Business/Commands/EventComment/CreateEventCommentCommand.cs:37]: empty line

File: src/EventService.Business/Commands/EventComment/CreateEventCommentCommand.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  {
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.ConvertAll(er => er.ErrorMessage));
+    }
+
+    DbEventComment dbEventComment = _mapper.Map(request);
```
Comment (VLiamin) [inline src/EventService.Business/Commands/EventComment/CreateEventCommentCommand.cs]: May be remove variable?

**Review by VLiamin — APPROVED** [review]

**Review by SolovyovDmitry — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**